NumLib/Extrapolation/LocalLinearLeastSquaresExtrapolator.h

```diff
@@ -0,0 +1,87 @@
+#ifndef PROCESSLIB_LOCAL_LLSQ_EXTRAPOLATOR_H
+#define PROCESSLIB_LOCAL_LLSQ_EXTRAPOLATOR_H
+
+#include "MathLib/LinAlg/GlobalMatrixProviders.h"
+#include "Extrapolator.h"
+
+namespace NumLib
+{
+
+/*! Extrapolator doing a linear least squares extrapolation locally in each
+ *  element.
+ *
+ * The results of the element-wise least squares are averaged at each node.
+ *
+ * \note
+ * \parblock
+ * The described procedure is not least-squares optimal on the whole mesh! But
+ * the residuals are computed from the actual interpolation result that is being
+ * returned by this class.
+ *
+ * Currently this class only supports interpolating single component variables.
+ * \endparblock
+ */
```

Comment:
add comment and assertion that number of gauss points must be less than number of nodes per element.